### PR TITLE
Allow to pass custom handlers and key to traverse.children

### PR DESCRIPTION
**Pull request by qfox**

### CODE REVIEW

File: packages/mdast-util-to-bemjson/lib/traverse.js [hunk 1]
```diff
@@ -12,15 +12,17 @@ const assert = require('assert');
  */
 function traverse(transform, node, parent) {
     const type = node && node.type;
-    const handler = transform.handlers[type] || transform.handlers.default;
-
     assert(type, `Expected node, got '${node}'`);
 
+    const baseHandler = transform.handlers[type] || transform.handlers.default;
+    const userHandler = transform.userHandlers[type];
+    const handler = userHandler ? userHandler.bind({ __base: baseHandler }) : baseHandler;
```
Comment (birhoff) [inline packages/mdast-util-to-bemjson/lib/traverse.js]: inherit костыль?

File: packages/mdast-util-to-bemjson/test/options.test.js [hunk 2]
```diff
@@ -24,5 +24,16 @@ describe('Test converter options', () => {
...
         });
+
+        it('should rewrite bundled handlers with passed ones via `handlers` option', () => {
+            const bjson = toBemjson({ type: 'text', value: 'testcode' }, { handlers: {
+                text(transform, node) {
+                    return transform(node, { block: node.value });
+                }
+            } });
+
+            console.log(bjson);
```
Comment (birhoff) [inline packages/mdast-util-to-bemjson/test/options.test.js]: remove

File: packages/mdast-util-to-bemjson/lib/traverse.js [hunk 3]
```diff
@@ -12,15 +12,17 @@ const assert = require('assert');
...
+
     return handler(transform, node, parent);
 }
 
-function traverseChildren(transform, parent) {
-    const nodes = parent.children || [];
+function traverseChildren(transform, parent, key) {
+    const nodes = parent['children' || key] || [];
```
Comment (birhoff) [inline packages/mdast-util-to-bemjson/lib/traverse.js]: maybe you want
```
let nodes = parent['children'] || parent[key]  || [];
if(!Array.isArray(nodes)){
nodes =  nodes.push(nodes)
// or
nodes = []
}
```
we should check that nodes are array

**Review by birhoff — CHANGES_REQUESTED** [review]
few things

File: packages/mdast-util-to-bemjson/lib/traverse.js (hunk 1 above)
Comment (qfox) [reply packages/mdast-util-to-bemjson/lib/traverse.js]: возможность фолбечиться на базовый

**Review by birhoff — APPROVED** [review]

**Pull request merged**